### PR TITLE
Add user to grroup

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
Add a User to a group with autosuggest

# What has changed
Added it

# How to test?
Fire it up, add some users and groups in support tool.  Then play in ROPS

# Links
https://trello.com/b/yarvSxGu/generic-response-management

### CODE REVIEW

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Screenshot 2022-01-14 at 11 04 28

The styling of the suggestions seems off to me, not clearly connected to the current search. Also they cannot currently be tabbed through, which may be an accessibility issue?

**Review by hbrace — APPROVED** [review]

File: ui/src/AddUserToGroup.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newuUserGroupMember),
+    });
+
+    if (response.ok) {
+      history.push(
+        `/groupadmin?groupId=${props.groupId}&groupName=${props.groupName}`
+      );
+      // history.push(`/surveys?flashMessageUntil=${Date.now() + 5000}`);
```
Comment (liamtoozer) [inline ui/src/AddUserToGroup.js:102]: Safe to remove commented out line?

File: ui/src/AddUserToGroup.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    if (userId === undefined) {
+      setErrorSummary([
+        "Please select a user to add by the autocomplete below",
+      ]);
+      setHasErrors(true);
+      addUserInProgress = false;
+      return;
+    }
+
+    const newuUserGroupMember = {
```
Comment (liamtoozer) [inline ui/src/AddUserToGroup.js:87]: Small typo

File: ui/src/AddUserToGroup.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+  }
+
+  async function addUserToGroup() {
+    if (addUserInProgress) {
+      return;
+    }
+
+    addUserInProgress = true;
+
+    setErrorSummary("");
```
Comment (liamtoozer) [inline ui/src/AddUserToGroup.js:73]: Set to empty list instead of empty string?

File: ui/src/AddUserToGroup.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
+import React, { useEffect, useRef, useState } from "react";
+import { useHistory } from "react-router-dom";
```
Comment (liamtoozer) [inline ui/src/AddUserToGroup.js:2]: useHistory and Link are both imported from same module, worth combining?

File: ui/src/AddUserToGroup.js [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+      setUserList(users);
+      setSuggestions(users);
+    }
+
+    fetchUsers();
+  }, []);
+
+  function cancel() {
+    history.push(
+      `/groupadmin?groupId=${props.groupId}&groupName=${props.groupName}`
```
Comment (AdamHawtin) [inline ui/src/AddUserToGroup.js:49]: I think needs proper URL encoding to be safe with the group name

**Review by liamtoozer — COMMENTED** [review]
Looks great so far and works for me, just a few things spotted by eye and linter

File: ui/src/AddUserToGroup.js [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Add User To Group</title>
+      </Helmet>
+      <Announcer text={"Add User Page Page"} />
+      <Link
+        to={`/groupadmin?groupId=${props.groupId}&groupName=${props.groupName}`}
```
Comment (AdamHawtin) [inline ui/src/AddUserToGroup.js:193]: Needs URL encoding

File: ui/src/AddUserToGroup.js [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+    const response = await fetch("/api/userGroupMembers", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newuUserGroupMember),
+    });
+
+    if (response.ok) {
+      history.push(
+        `/groupadmin?groupId=${props.groupId}&groupName=${props.groupName}`
```
Comment (AdamHawtin) [inline ui/src/AddUserToGroup.js:100]: Needs URL encoding

File: ui/src/AddUserToGroup.js [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    value,
+    onChange: onChange,
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Add User To Group</title>
+      </Helmet>
+      <Announcer text={"Add User Page Page"} />
```
Comment (AdamHawtin) [inline ui/src/AddUserToGroup.js:191]: Page page?

File: ui/src/AddUserToGroup.js [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+      <Link
+        to={`/groupadmin?groupId=${props.groupId}&groupName=${props.groupName}`}
+      >
+        ← Back to group admin
+      </Link>
+      {errorSummary.length > 0 && (
+        <ErrorSummary errorSummary={errorSummary} ref={errorSummaryTitle} />
+      )}
+      <h1>Add User To Group {props.groupName} Page</h1>
```
Comment (AdamHawtin) [inline ui/src/AddUserToGroup.js:200]: Should just be  "Add User To Group ..."

**Comment by LukeLoze1** [comment]
> Screenshot 2022-01-14 at 11 04 28
> 
> The styling of the suggestions seems off to me, not clearly connected to the current search. Also they cannot currently be tabbed through, which may be an accessibility issue?
> 
> Apart from tabbing though, the functionality seems to work very nicely.

Fixed by moving CSS file

File: ui/src/AddUserToGroup.js (hunk 3 above)
Comment (LukeLoze1) [reply ui/src/AddUserToGroup.js:75]: made it a string when declaring too

**Review by liamtoozer — APPROVED** [review]
LGTM, great work 👍

**Pull request merged**